core/trino-main/src/main/java/io/trino/operator/scalar/ArrayVectorFunctions.java

```diff
@@ -56,6 +56,14 @@ public static double dotProduct(@SqlType("array(double)") Block first, @SqlType(
         return dotProduct;
     }
 
+    @Description("Calculates the cosine similarity between two vectors")
+    @ScalarFunction
+    @SqlType(StandardTypes.DOUBLE)
+    public static double cosineSimilarity(@SqlType("array(double)") Block first, @SqlType("array(double)") Block second)
+    {
+        return 1.0 - cosineDistance(first, second);
```

Comment:
I'd do it the other way around. Implement this method using the logic in cosineDistance (except for the bit at the end), and make cosineDistance = 1 - cosineSimilarity.